### PR TITLE
Path deform tool update.

### PR DESCRIPTION
**Pull request by seagetch**
Added bunch of changes to path deform tool.

1. Partial deformation for selected vertices.
     Path deformation tool can select vertices when pressing Alt key. When applying deformation, only selected vertices are moved.

2. Update undo / redo of path deformation tool.
     Redo / undo of Path deformation tool remembers the keypoint of the mesh editor.

3. Partial reset / flipping
     When selecting vertices, right click menu shows "Reset selected" and "Flip selected horizontally from mirror" on the menu. It applies the reset / flip of the deformation only to the selected vertices.

### CODE REVIEW

File: source/creator/utils/transform.d [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+    * 3. copy deformation from other parts.
+    *    extrapolation = false, axis = 2
+    * Params:
+    *   binding = destination binding
+    *   srcBinding = source binding to be copied
+    *   index = target index of binding
+    *   axis = 0: flip horizontally, 1: flip vertically, -1: flip diagonally, 2: not flipped
+    *   extrapolation = specifying source index is selected in mirroered position or not.
+    */
+void autoFlipBinding(ParameterBinding binding, ParameterBinding srcBinding, vec2u index, uint axis, bool extrapolation = true, ulong[]* selected = null) {
```
Comment (LunaTheFoxgirl) [inline source/creator/utils/transform.d]: Freestanding functions should use the inc\<Space\>\<Function Name\> convention.

In this case incBindingAutoFlip

File: source/creator/utils/transform.d [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+import std.string;
+import inochi2d;
+import i18n;
+import std.stdio;
+import creator.utils;
+import std.algorithm.searching: countUntil;
+
+
+
+ParameterBinding getPairBindingFor(Parameter param, Node target, FlipPair pair, string name, bool forceCreate = false) {
```
Comment (LunaTheFoxgirl) [inline source/creator/utils/transform.d]: See autoFlipBinding,

incBindingGetPairFor, would be appropriate here

**Review by LunaTheFoxgirl — CHANGES_REQUESTED** [review]
We should keep the code style consistent, otherwise this commit is almost ready to merge.

File: source/creator/utils/transform.d (hunk 1 above)
Comment (seagetch) [reply source/creator/utils/transform.d]: renamed autoFlipBinding to incBindingAutoFlip.

File: source/creator/utils/transform.d (hunk 2 above)
Comment (seagetch) [reply source/creator/utils/transform.d]: renamed getPairBindingFor to incBindingGetPairFor.

**Review by LunaTheFoxgirl — APPROVED** [review]
LGTM, merging

**Pull request merged**